### PR TITLE
fix!: Rename package to match name inferred from git URL

### PR DESCRIPTION
**Pull request by jbelkins**
## Issue \#
Fixes https://github.com/awslabs/aws-sdk-swift/issues/680

## Description of changes
Change the declared name of this package to match its inferred name (i.e. the last path component of its Git URL, minus the extension).  This is being performed in conjunction with similar renames of `aws-sdk-swift` and `smithy-swift`.  A complete rationale for making this change is given in [this PR](https://github.com/awslabs/aws-sdk-swift/pull/678).

**This is a breaking change** but consumers of this package simply need to update their `Package.swift` to refer to the package by the new name.

Specific changes:
- Rename package from `ClientRuntime` to `smithy-swift`
- Use new name & modern syntax when generating manifests

## Scope
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ fun writePackageManifest(settings: SwiftSettings, fileManifest: FileManifest, de
                     writer.write("name: \"${settings.moduleName}Tests\",")
                     writer.openBlock("dependencies: [", "],") {
                         writer.write("\$S,", settings.moduleName)
-                        writer.write(".product(name: \"SmithyTestUtil\", package: \"ClientRuntime\")")
```
[the remark this answers is not in the document]
Comment (ganeshnj) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt:58]: Few things must be verified
- Individual package build located at /Users/jangirg/Projects/Amplify/SwiftSDK/aws-sdk-swift/codegen/sdk-codegen/build/smithyprojections/sdk-codegen/{service}/swift-codegen
- when packages are staged in  `release`
- opening aws-crt-swift in xcode
- opening smityh-swift in xcode
- opening aws-sdk-swift in xcode without staged changes
- opening aws-sdk-swift in xcode with staged changes

[the remark this answers is not in the document]
Comment (jbelkins) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt:58]: The broken `Package.swift` above is fixed in https://github.com/awslabs/aws-sdk-swift/pull/678/commits/f290fd48c428584dc7f509be83a074efe4d2d2b6

[the remark this answers is not in the document]
Comment (jbelkins) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt:58]: @ganeshnj All the situations you described have been tested out.  I generated a release then built out all services as well.

[the remark this answers is not in the document]
Comment (ganeshnj) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt:58]: Thanks for testing. This is painful but needed until we have a better CI checks for such cases.

**Review by ganeshnj — APPROVED** [review]

**Pull request merged**